src/main.zig

```diff
@@ -190,10 +190,10 @@ pub fn serialize(comptime T: type, data: T, l: *ArrayList(u8)) !void {
             if (info.Union.tag_type == null) {
                 return error.UnionIsNotTagged;
             }
-            inline for (info.Union.fields) |f, index| {
+            inline for (info.Union.fields, 0..) |f, index| {
```

Comment:
Now if you want the index, you have to explicitly iterate in a range that will automatically detect the `END` from the other list. 

I think the logic for this breaking change is to have a less-magical API.